File: java/src/main/java/P179.java
import java.util.stream.IntStream;

@SuppressWarnings("unused")
public class P179 implements Puzzle {

    final int limit = 10_000_000;

    @Override
    public String solve() {

        int divisors[] = new int[limit + 1];

        IntStream.range(1, limit + 1)
                .parallel()
                .forEach(i -> divisors[i] = (int) Util.divisors(i).count());

        int count = 0;
        for (int i = 2; i < limit; i++) {
            if (divisors[i] == divisors[i + 1]) {
                count++;
            }
        }

        return "" + count;
    }
}

File: c/p179.c
#include <stdio.h>

int divisors(const int n);

int main(int argc, char** argv) {
    const int limit = 10000000;
    int last = 0;
    int count = 0;
    for (int i = 2; i <= limit; ++i) {
        const int d = divisors(i);
        if (d == last) {
            ++count;
        }
        last = d;
    }
    printf("%d\n", count);
}


int divisors(const int n) {
    int count = 2;
    int d = 2;
    for (; d * d <= n; ++d) {
        if (n % d == 0) {
            count += 2;
        }
    }
    --d;
    if (d * d == n) {
        --count;
    }

    return count;
}

